### PR TITLE
Stricter merge-with

### PR DESCRIPTION
**Pull request by borkdude**

### CODE REVIEW

File: src/speculative/core.cljc [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 (defn map-entry? [v]
   #?(:cljs (clojure.core/map-entry? v)
-     :clj #(instance? java.util.Map$Entry %)))
+     :clj (instance? java.util.Map$Entry v)))
```
Comment (mfikes) [inline src/speculative/core.cljc]: This will fail for entries from Clojure's sorted maps (and could theoretically fail for other map implementations).

**Comment by borkdude** [comment]
@mfikes I'm not sure why we used a custom `map-entry?` predicate, but this is already in both clj + cljs, so I removed that. I believe your request for change was about this predicate. Could you look again?

**Comment by borkdude** [comment]
One note:

```
$ clj -A:test:repl
user=> (stest/instrument `merge-with)
[clojure.core/merge-with]
user=> (merge-with + {:a 1} [[:a 2]])
Evaluation error - invalid arguments to clojure.core/merge-with at (NO_SOURCE_FILE:2).
[:a 2] - failed: map-entry? at: [:maps] spec: :speculative.core/map-entry
[:a 2] - failed: map-entry? at: [:maps] spec: :speculative.core/map-entry
```
I'm not sure why spec prints the error about `[:a 2]` twice here.

**Comment by borkdude** [comment]
Please wait a moment before merging, I want to look into the above printing issue.

**Pull request merged**